src/main/java/frc/robot/Diagnostics.java

```diff
@@ -0,0 +1,39 @@
+package frc.robot;
+
+import edu.wpi.first.math.kinematics.ChassisSpeeds;
+import edu.wpi.first.wpilibj2.command.Command;
+import frc.robot.subsystems.drive.Drive;
+import frc.robot.subsystems.indexer.Indexer;
+import frc.robot.subsystems.pivot.Pivot;
+import frc.robot.subsystems.shooter.Shooter;
+
+public class Diagnostics {
+
+  private Diagnostics() {}
+
+  // run drivebase
+  public static Command testDrivetrainCommand(Drive drive) {
+    return drive.runVelocityCommand(() -> new ChassisSpeeds(1, 1, 1));
+  }
+
+  // run shooter
+  public static Command testShooterCommand(Shooter shooter) {
+    return shooter.setVelocityRPMCommand(60);
+  }
+
+  // run pivot
+  public static Command testPivotCommand(Pivot pivot) {
+    return pivot
+        .goToAngle(Constants.PivotConstants.PIVOT_MAX_POS_RAD)
+        .andThen(pivot.goToAngle(Constants.PivotConstants.PIVOT_MIN_POS_RAD));
+  }
+
+  // run indexer
+  public static Command testIndexerCommand(Indexer indexer) {
+    return indexer.indexerOnIntakeOnCommand()
+    .andThen(indexer.indexerOffIntakeOffCommand());
+  }
```

Comment:
this will run both of them and then immediately stop because the commands are runOnce, you should add a wait command in between

Reply:
nvm all of indexer is bugged lol, can you change all of the `runOnce` commands in `Indexer` to be `run` and then you can use `command.withTimeout` to run for a specific period of time